### PR TITLE
Add more cases to keras _convert_reshape

### PR DESCRIPTION
**Pull request by apivovarov**
Currently Keras `_convert_reshape` only supports cases where Keras layer input channel and target shape channel are equal.
e.g. ?,4,4,256 -> ?,16,256

This PR supports additional cases 
1. target shape length is 1
?,2,8 -> ?,16

2. target shape length is 2 and shape is square, e.g.
?,2,8 -> ?,4,4
?,16 -> ?,4,4

I use this improvement in Keras `_convert_reshape` to support `Softmax(axis=[1,2])` (global Softmax).
Relay does not support array type for  Softmax axis (e.g. `axis=[1,2]`). This is why I reshape Softmax input to single axis and then reshape Softmax output back.
Subgraph example: https://www.dropbox.com/s/51rp0g7oul8vg4e/Screenshot%202019-08-27%2022.32.16.png?dl=0

Keras example:
```
#!/usr/bin/env python3
import keras
from keras.models import Sequential
from keras.optimizers import Nadam
from keras.layers import Dense, Activation, Input, Reshape, Softmax
import numpy as np

inputs = Input(shape=(8,8,4,))
l2=Reshape((16,16))(inputs)
outputs=Softmax(axis=[1,2])(l2)
model = keras.Model(inputs, outputs)
model.compile(optimizer=Nadam(), loss='categorical_crossentropy', metrics=['accuracy'])
model.summary()

dshape=(1,8,8,4)
x = np.full(dshape, 0.9903)
preds = model.predict(x)

print(preds)
print("Sum(output): {}".format(np.sum(preds)))
```

### CODE REVIEW

**Comment by apivovarov** [comment]
@Huyuwei Can you merge it please?
